feature(BaseMonitorSet): Implement monitoring template

k0machi opened the pull request:
This commit adds a new templating mechanism for monitoring dashboards,
allowing SCT to be in sync with changes in scylla-monitoring repo, by
first separating SCT metrics into a separate file, and then injecting
them into the overview template on the monitoring node, then
regenerating said template. This allows monitoring dashboard to always
be up to date with regards to changes on the monitoring side.

Fixes scylladb/qa-tasks#1444

### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [x] [Jenkins](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/alexey/job/alexey-argus-testing/108/)

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

k0machi commented:
Demo:

![image](https://github.com/user-attachments/assets/78543d7b-e9ec-4b4b-be48-1940619ed560)


One more thing that's missing are previously available template variables for event filtering - I'll add them soon.

soyacz commented on sdcm/cluster.py:
```diff
@@ -5647,6 +5648,45 @@ def download_scylla_monitoring(self, node):
+        def find_overview_row(row):
+            return row["class"] == "row" and len(row.get("panels", [])) > 0 and row["panels"][0].get("class", "") == "alert_table"
+
+        template_file = f"/tmp/scylla-overview.template.{node.name}.json"
```
for adjusting remote files like that we actually have ctx mgr: `remote_file` did you check if could be reused?

k0machi replied:
Fixed

k0machi commented:
Added annotations, template variables for event types, added logic to change default values (currently just hardcoded to change the `by` var, open for suggestions on the format if we want something more generic), re-arranged widgets to be more in line with the previous dash.

k0machi commented:
[This run](https://argus.scylladb.com/tests/scylla-cluster-tests/a197e284-70a0-4997-a2e3-e0356f067f29) has the updated dashboard, monitoring node is set to `keep` so will be alive for now.

soyacz commented:
requests served per instance shows whole numbers (like 30000 instead of 30k) and miss unit (ops/s)

k0machi commented:
> requests served per instance shows whole numbers (like 30000 instead of 30k) and miss unit (ops/s)

Fixed

k0machi commented:
I'll update the template based on recent changes by Valerii and then we can merge this.

soyacz reviewed (approved):
LGTM
Still docker provision test fails with c-s.
@dimakr do you know what could be the reason?

fruch commented:
> LGTM Still docker provision test fails with c-s. @dimakr do you know what could be the reason?

it's cause of counters, we need to drop them from those tests.

since we move to take latest scylla release (with tablets enabled by default), it's kind of broken
and forgot/didn't got to handle/report it

k0machi commented:
Will rebase onto #10006 and update with changes from recent stress-command panels merge.

k0machi commented:
[Latest run](https://argus.scylladb.com/tests/scylla-cluster-tests/c565dd6d-824d-4b0f-979b-c9d6360cf12f) that implements the new dash. Grafana node is still up.

fruch reviewed (approved):
LGTM
